### ISSUE TITLE
`setLatLngBounds` is poorly named

### ISSUE THREAD

**Issue by 1ec5** [maplibre/maplibre-native#1513]
#1455 adds the methods `-[MLNMapView setLatLngBounds:]` and `-[MLNMapView clearLatLnBounds]`, which are poorly named. Effectively, the iOS/macOS SDK now has a method named after the Android SDK’s `LatLngBounds` class, which makes little sense. That class is analogous to the `MLNCoordinateBounds` struct on iOS/macOS, so it should be called something like `setCoordinateBounds`.

Better yet, it should be modeled as a property called something more descriptive like `viewableCoordinateBounds`. As it is, this is a write-only property, nearly unprecedented in any Objective-C framework. “Clearing” the coordinate bounds could be tricky because `MLNCoordinateBounds` is a struct, which can’t nullable (optional) in Objective-C. Typically, this library has dealt with that issue by allowing the property to be set to a magic value of some kind. In this case, MLNGeometry.h could define a constant `MLNCoordinateBoundsWorld` that encompasses the entire coordinate system.

Regardless, there should not be any symbol with “LatLngBounds” in its name. This violates multiple Cocoa naming conventions, including the prohibition on superfluous abbreviations.

**Comment by louwers** [comment]
Thanks for sharing your perspective. I think the naming comes from the internal C++ API that is called, not the Android SDK. That said your arguments for changing the API make sense and I would be happy to merge an alternate API before the next release if someone make a PR for it.

Related issue which argues to smooth over the differences between the platform SDKs https://github.com/maplibre/maplibre-native/issues/1462

**Comment by 1ec5** [comment]
> I think the naming comes from the internal C++ API that is called, not the Android SDK.

You’re right that it’s a little more complex than I described earlier. `mbgl::LatLngBounds` was originally called `BoundingBox` but was renamed to `LatLngBounds` in mapbox/mapbox-gl-native@78cbab950731ece92b467f1e3230ebb13f1ae7b4 as part of mapbox/mapbox-gl-native#1018 mapbox/mapbox-gl-native#1021. Consistency with GL JS was given as the reason. GL JS got the term from Leaflet (mapbox/mapbox-gl-js#369), and Leaflet probably got the term from [Google Maps](https://developers.google.com/maps/documentation/javascript/reference/coordinates#LatLngBounds).[^android] However, the `LatLngBounds` terminology was never exposed to external developers until the Android SDK later renamed `CoordinateBounds` to `LatLngBounds`, again for consistency with Google Maps: https://github.com/mapbox/mapbox-gl-native/issues/2896#issuecomment-153505223.

So in short, this method is named after something in Google Maps. Consistency with other platforms is one thing, but this method is very inconsistent with other parts of _this_ library and the language it’s written in. @JulianBissekkou, given your work on #1455, would you be open to revisiting that API design?

[^android]: At the time, the Android SDK still called it `BoundingBox`. That was introduced in mapbox/mapbox-gl-native#878, but you can follow the history all the way back to tobrun/mapbox-android-sdk@4913abb61d5fb7c095d070a75f5ef2a8027d3b4c osmdroid/osmdroid@c0449add2e7e0fc5fe8b11e5eb989413baea99bc. At some point, a redundant `CoordinateBounds` type was introduced, but it was removed for mapbox/mapbox-gl-native#2896.

**Comment by louwers** [comment]
This didn't make the cut because there was a typo in the method name.

A PR with another name is welcome.
